tests/unit/multiclass/BaggingMachine_unittest.cc

```diff
@@ -230,11 +267,78 @@ TEST(BaggingMachine,classify_CART)
 	EXPECT_EQ(1.0,res_vector[3]);
 	EXPECT_EQ(1.0,res_vector[4]);
 
-	CMulticlassAccuracy* eval=new CMulticlassAccuracy();
+	auto eval = some<CMulticlassAccuracy>();
 	EXPECT_NEAR(0.642857,c->get_oob_error(eval),1e-6);
 
-	SG_UNREF(test_feats);
 	SG_UNREF(result);
-	SG_UNREF(c);
-	SG_UNREF(eval);
+}
+
+TEST_F(BaggingMachine, output_binary)
+{
+	CCARTree* cart=new CCARTree();
+	CMeanRule* cv=new CMeanRule();
+
+	cart->set_feature_types(ft);
+	auto c = some<CBaggingMachine>(features_train,labels_train);
+	c->parallel->set_num_threads(1);
+	c->set_machine(cart);
+	c->set_bag_size(14);
+	c->set_num_bags(10);
+	c->set_combination_rule(cv);
+	c->train(features_train);
+
+	CBinaryLabels* result=c->apply_binary(features_test);
+	SGVector<float64_t> res_vector=result->get_labels();
+	SGVector<float64_t> values_vector=result->get_values();
+
+	EXPECT_EQ(1.0,res_vector[0]);
+	EXPECT_EQ(-1.0,res_vector[1]);
+	EXPECT_EQ(-1.0,res_vector[2]);
+	EXPECT_EQ(1.0,res_vector[3]);
+	EXPECT_EQ(1.0,res_vector[4]);
+
+	EXPECT_DOUBLE_EQ(1.0,values_vector[0]);
+	EXPECT_DOUBLE_EQ(0.3,values_vector[1]);
+	EXPECT_DOUBLE_EQ(0.3,values_vector[2]);
+	EXPECT_DOUBLE_EQ(1.0,values_vector[3]);
+	EXPECT_DOUBLE_EQ(0.7,values_vector[4]);
+
+	SG_UNREF(result);
+}
+
+TEST_F(BaggingMachine, output_multiclass)
+{
+
+	CCARTree* cart=new CCARTree();
+	CMeanRule* cv=new CMeanRule();
+
+	cart->set_feature_types(ft);
+	auto c = some<CBaggingMachine>(features_train,labels_train);
+	c->parallel->set_num_threads(1);
+	c->set_machine(cart);
+	c->set_bag_size(14);
+	c->set_num_bags(10);
+	c->set_combination_rule(cv);
+	c->train(features_train);
+
+	CMulticlassLabels* result=c->apply_multiclass(features_test);
+
+	SGVector<float64_t> res_vector=result->get_labels();
+	SGVector<float64_t> values_vector=result->get_values();
+
+	EXPECT_EQ(1.0,res_vector[0]);
+	EXPECT_EQ(0.0,res_vector[1]);
+	EXPECT_EQ(0.0,res_vector[2]);
+	EXPECT_EQ(1.0,res_vector[3]);
+	EXPECT_EQ(1.0,res_vector[4]);
+
+  int32_t num_labels = result->get_num_labels();
+
+  for (int32_t i = 0; i < num_labels; ++i)
+  {
+    SGVector<float64_t> confidences = result->get_multiclass_confidences(i);
+    EXPECT_DOUBLE_EQ(1.0, linalg::sum(confidences));
```

Comment:
such assertions should appear in the test name (if you write the test yourself)
i.e. output_mutliclass_probs_sum_to_one